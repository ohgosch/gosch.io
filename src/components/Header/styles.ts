import styled, { css } from 'styled-components';
import { hideVisually, rem } from 'polished';
import { mobile } from 'visual/medias';

export const Container = styled.header`
  ${({ theme }) => css`
    height: ${rem(theme.sizes.headerHeight)};
    display: flex;
    justify-content: space-between;
    align-items: center;

    ${mobile(css`
      justify-content: center;
    `)}
  `}
`;

export const Title = styled.h1`
  ${hideVisually()}
`;

export const LogoLink = styled.a`
  display: block;

  svg {
    height: ${rem(45)};
  }
`;

export const Menu = styled.nav`
  display: flex;
  gap: ${rem(16)};
  align-items: center;

  ${mobile(css`
    ${hideVisually()}
  `)}
`;

export const MenuItem = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    padding: ${rem(8)} ${rem(16)};
    transition: 0.2s ease;
    cursor: pointer;
    text-decoration: none;
    text-transform: lowercase;

    :not(.highlight) {
      :after {
        content: '';
        transition: 0.2s ease;
        display: block;
        height: ${rem(1)};
        background-color: ${theme.colors.white};
        width: 0;
      }

      :hover:after {
        width: 100%;
      }
    }

    &.highlight {
      border: ${rem(2)} solid ${theme.colors.primary};
      border-radius: ${rem(18)};

      :hover {
        background-color: ${theme.colors.primary};
        color: ${theme.colors.black};
      }
    }
  `}
`;
