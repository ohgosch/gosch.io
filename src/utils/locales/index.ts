import I18nextBrowserLanguageDetector from 'i18next-browser-languagedetector';
import I18NextLocizeBackend from 'i18next-locize-backend';
import { initReactI18next } from 'react-i18next';
import i18n from 'i18next';

import ptBR from './ptBR.json';
import en from './en.json';

i18n
  .use(I18NextLocizeBackend)
  .use(initReactI18next)
  .use(I18nextBrowserLanguageDetector)
  .init({
    debug: false,
    fallbackLng: 'en',
    whitelist: ['pt', 'en'],
    supportedLngs: ['pt', 'en'],
    detection: {
      order: ['querystring', 'navigator'],
      lookupQuerystring: 'lng',
    },
    resources: {
      en: {
        translation: en,
      },
      pt: {
        translation: ptBR,
      },
    },
  });

export default i18n;
