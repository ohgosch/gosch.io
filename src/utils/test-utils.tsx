import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { ReactElement } from 'react';

import theme from 'visual/theme';

const customRender = (ui: ReactElement) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);

export * from '@testing-library/react';
export { customRender as render };
