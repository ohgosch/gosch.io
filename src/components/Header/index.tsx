import Link from 'next/link';

import Logo from 'public/assets/img/logo.svg';

import * as S from './styles';

const Header = () => {
  return (
    <S.Container>
      <S.Title>Gosch</S.Title>
      <S.LogoLink href="/" title="Go home">
        <Logo aria-hidden />
      </S.LogoLink>
      <S.Menu>
        <Link href="#services" passHref>
          <S.MenuItem>Services</S.MenuItem>
        </Link>
        <Link href="#contact" passHref>
          <S.MenuItem className="highlight">Contact</S.MenuItem>
        </Link>
      </S.Menu>
    </S.Container>
  );
};

export default Header;
