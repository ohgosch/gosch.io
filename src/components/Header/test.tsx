import { render, screen } from 'utils/test-utils';

import Header from '.';

describe('<Header />', () => {
  it('should render title', () => {
    const { container } = render(<Header />);

    expect(screen.getByRole('heading', { name: /Gosch/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
