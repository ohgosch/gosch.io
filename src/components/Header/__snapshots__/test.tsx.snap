// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render title 1`] = `
.c0 {
  height: 6.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c2 {
  display: block;
}

.c2 svg {
  height: 2.8125rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  color: #F2F2F2;
  padding: 0.5rem 1rem;
  -webkit-transition: 0.2s ease;
  transition: 0.2s ease;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: lowercase;
}

.c4:not(.highlight):after {
  content: '';
  -webkit-transition: 0.2s ease;
  transition: 0.2s ease;
  display: block;
  height: 0.0625rem;
  background-color: #F2F2F2;
  width: 0;
}

.c4:not(.highlight):hover:after {
  width: 100%;
}

.c4.highlight {
  border: 0.125rem solid #DBA14B;
  border-radius: 1.125rem;
}

.c4.highlight:hover {
  background-color: #DBA14B;
  color: #171717;
}

@media (max-width:46rem) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:46rem) {
  .c3 {
    border: 0;
    -webkit-clip: rect(0 0 0 0);
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    white-space: nowrap;
    width: 1px;
  }
}

<header
  class="c0"
>
  <h1
    class="c1"
  >
    Gosch
  </h1>
  <a
    class="c2"
    href="/"
  >
    <svg
      aria-hidden="true"
      fill="none"
      height="45"
      width="145"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M52.99 29.283a10.039 10.039 0 0 1-3.639-3.695 10.198 10.198 0 0 1-1.331-5.041c0-1.77.459-3.508 1.331-5.041a10.038 10.038 0 0 1 3.639-3.695 9.705 9.705 0 0 1 5.008-1.347c.414 0 .827.028 1.237.085a.524.524 0 0 1 .383.224.539.539 0 0 1 .082.44l-.442 2.42a.502.502 0 0 1-.355.427.487.487 0 0 1-.194.015l-.711-.028a6.186 6.186 0 0 0-3.209.876 6.452 6.452 0 0 0-2.325 2.354 6.536 6.536 0 0 0 0 6.502 6.42 6.42 0 0 0 2.325 2.354c.972.58 2.081.883 3.209.876a5.6 5.6 0 0 0 3.51-1.154 6.69 6.69 0 0 0 2.224-3.07h-3.54a.515.515 0 0 1-.385-.164.545.545 0 0 1-.163-.391v-2.024a.569.569 0 0 1 .164-.39.554.554 0 0 1 .385-.166h5.845c.352-.02.702.057 1.014.221a.712.712 0 0 1 .358.64 10.782 10.782 0 0 1-1.21 5.08 9.31 9.31 0 0 1-3.361 3.654 8.923 8.923 0 0 1-4.841 1.347 9.704 9.704 0 0 1-5.008-1.309zm20.028-2.053a9.968 9.968 0 0 1-1.99-3.418l-.051-.141a9.835 9.835 0 0 1-.498-3.108c.003-1.154.198-2.3.577-3.39a9.477 9.477 0 0 1 1.73-3.027 12.768 12.768 0 0 1 3.06-2.514 8.91 8.91 0 0 1 3.91-1.003h.08a9.568 9.568 0 0 1 4.5.829 9.832 9.832 0 0 1 2.646 1.793 8.277 8.277 0 0 1 1.86 2.354 9.893 9.893 0 0 1 1.302 4.911 10.22 10.22 0 0 1-.465 3.042 9.983 9.983 0 0 1-1.395 2.764 10.418 10.418 0 0 1-1.492 1.727c-.608.536-1.276.998-1.99 1.375a9.604 9.604 0 0 1-6.264.97 9.68 9.68 0 0 1-5.52-3.15v-.014zm11.95-2.476c.44-.488.8-1.043 1.07-1.644a6.58 6.58 0 0 0 .55-2.594 6.395 6.395 0 0 0-.991-3.432 6.007 6.007 0 0 0-2.716-2.354 6.382 6.382 0 0 0-2.604-.556 6.097 6.097 0 0 0-4.315 1.747l-.466.504a6.576 6.576 0 0 0-1.11 1.968 6.577 6.577 0 0 0 .204 4.85 5.852 5.852 0 0 0 1.674 2.18 6.26 6.26 0 0 0 6.906.82 6.889 6.889 0 0 0 1.799-1.49zm11.161 4.708a8.462 8.462 0 0 1-3.115-1.526.579.579 0 0 1-.163-.419c.019-.244.094-.481.218-.692.223-.39.467-.767.73-1.13a.571.571 0 0 1 .465-.277.886.886 0 0 1 .247.056A11.226 11.226 0 0 0 100.435 27a4.27 4.27 0 0 0 2.209-.471 1.452 1.452 0 0 0 .814-1.304 2.372 2.372 0 0 0-.577-1.413 2.595 2.595 0 0 0-1.139-.791 18.745 18.745 0 0 0-1.958-.654c-.093 0-.33-.085-.716-.194-1.232-.33-2.4-.87-3.455-1.596a4.434 4.434 0 0 1-1.646-2.354 5.931 5.931 0 0 1-.302-1.78 4.79 4.79 0 0 1 .739-2.594 5.77 5.77 0 0 1 2.14-1.958 8.567 8.567 0 0 1 3.291-.975c.168 0 .424-.024.772-.024 1.401.003 2.788.276 4.088.806 1.423.721 2.054 1.332 1.892 1.831-.165.37-.378.717-.632 1.031-.321.42-.525.692-.618.82a.278.278 0 0 1-.247.113 2.551 2.551 0 0 1-.93-.335 6.446 6.446 0 0 0-1.46-.527 10.11 10.11 0 0 0-2.06-.278 5.577 5.577 0 0 0-1.781.306 3.219 3.219 0 0 0-1.098.848c-.24.282-.376.64-.385 1.012a1.663 1.663 0 0 0 .772 1.36 9.265 9.265 0 0 0 1.822 1.032c.647.277 1.395.565 2.181.861.878.314 1.586.614 2.125.9a3.8 3.8 0 0 1 1.33 1.068c.417.425.734.938.93 1.502.239.6.365 1.238.372 1.884a4.486 4.486 0 0 1-.823 2.697 5.328 5.328 0 0 1-2.278 1.761 8.373 8.373 0 0 1-3.348.627 14.255 14.255 0 0 1-4.34-.72l.01-.029zm18.791-.301a9.983 9.983 0 0 1-3.552-3.667 10.134 10.134 0 0 1-1.298-4.966c0-1.741.447-3.452 1.298-4.966a9.983 9.983 0 0 1 3.552-3.667 9.266 9.266 0 0 1 4.874-1.346 9.492 9.492 0 0 1 3.92.833 9.38 9.38 0 0 1 3.255 2.387.526.526 0 0 1 .163.363.548.548 0 0 1-.191.358c-.386.315-.772.645-1.153.988l-.795.707a.528.528 0 0 1-.386.17.497.497 0 0 1-.358-.17 5.981 5.981 0 0 0-2.011-1.476 5.914 5.914 0 0 0-2.43-.52 5.759 5.759 0 0 0-3.06.856 6.333 6.333 0 0 0-2.237 2.307 6.5 6.5 0 0 0 0 6.333 6.3 6.3 0 0 0 2.237 2.307 5.764 5.764 0 0 0 3.06.862 5.676 5.676 0 0 0 2.274-.471 6.201 6.201 0 0 0 1.976-1.366.95.95 0 0 1 .605-.22.576.576 0 0 1 .437.169c.186.145.441.362.772.65.33.287.609.54.846.767a.751.751 0 0 1 .302.607.534.534 0 0 1-.163.39 9.606 9.606 0 0 1-5.653 3.009 9.546 9.546 0 0 1-6.27-1.228h-.014zm16.313.97a.377.377 0 0 1-.111-.278V11.184a.387.387 0 0 1 .111-.277.39.39 0 0 1 .275-.109h2.878a.353.353 0 0 1 .274.123c.069.07.109.165.112.263v7.251a.667.667 0 0 0 .19.448.65.65 0 0 0 .442.193h5.213a.589.589 0 0 0 .423-.18.728.728 0 0 0 .205-.47v-7.242a.395.395 0 0 1 .112-.263.356.356 0 0 1 .274-.123h2.878a.369.369 0 0 1 .359.236.369.369 0 0 1 .027.15v18.697a.376.376 0 0 1-.111.278.366.366 0 0 1-.275.113h-2.887a.346.346 0 0 1-.275-.127.388.388 0 0 1-.111-.264v-6.446a.728.728 0 0 0-.205-.47.565.565 0 0 0-.423-.18h-5.213a.598.598 0 0 0-.428.165.69.69 0 0 0-.204.443v6.488a.394.394 0 0 1-.112.264.34.34 0 0 1-.274.127h-2.879a.365.365 0 0 1-.274-.113l.009-.028zm-89.835-8.014V2.84a.823.823 0 0 0-.242-.6.805.805 0 0 0-.6-.23h-6.16a.737.737 0 0 0-.585.264.846.846 0 0 0-.236.566v15.482c-.027.37-.184.718-.442.981a1.242 1.242 0 0 1-.908.385h-6.238a1.202 1.202 0 0 0-.821.358 1.232 1.232 0 0 0-.355.83v4.33c.01.313.136.61.355.831.218.221.512.35.821.358h7.567a14.368 14.368 0 0 1-4.754 6.558 12.025 12.025 0 0 1-7.505 2.46 13.265 13.265 0 0 1-6.864-1.868 13.781 13.781 0 0 1-5.016-5.073 13.935 13.935 0 0 1-1.848-6.939c0-2.437.638-4.831 1.848-6.939a13.78 13.78 0 0 1 5.016-5.072 13.264 13.264 0 0 1 6.864-1.869l1.54.057c.279.026.557-.06.773-.24.216-.18.354-.439.382-.72l.94-5.166a1.162 1.162 0 0 0-.182-.933 1.139 1.139 0 0 0-.814-.474A19.202 19.202 0 0 0 21.294 0c-2.8-.014-5.575.534-8.164 1.612a21.177 21.177 0 0 0-6.923 4.665 21.45 21.45 0 0 0-4.613 7.001A21.643 21.643 0 0 0 0 21.536a21.62 21.62 0 0 0 2.812 10.767 21.314 21.314 0 0 0 7.774 7.887 20.731 20.731 0 0 0 10.709 2.902c3.65.052 7.239-.945 10.35-2.876a19.88 19.88 0 0 0 7.187-7.788 22.812 22.812 0 0 0 2.566-10.103v-.208z"
        fill="#F2F2F2"
      />
    </svg>
  </a>
  <nav
    class="c3"
  >
    <a
      class="c4"
      href="/#services"
    >
      Services
    </a>
    <a
      class="c4 highlight"
    >
      Contact
    </a>
  </nav>
</header>
`;
